File: apps/query-burst-msa/order-service/src/main/java/com/booster/queryburstmsa/order/web/dto/OrderResponse.java
package com.booster.queryburstmsa.order.web.dto;

import com.booster.queryburstmsa.order.domain.OrderStatus;
import com.booster.queryburstmsa.order.domain.entity.OrderEntity;

import java.time.LocalDateTime;
import java.util.List;

public record OrderResponse(
        Long orderId,
        Long memberId,
        OrderStatus status,
        long totalAmount,
        String reservationId,
        LocalDateTime orderedAt,
        List<OrderItemView> items
) {
    public static OrderResponse from(OrderEntity order) {
        return new OrderResponse(
                order.getId(),
                order.getMemberId(),
                order.getStatus(),
                order.getTotalAmount(),
                order.getReservationId(),
                order.getOrderedAt(),
                order.getItems().stream()
                        .map(item -> new OrderItemView(item.getProductId(), item.getCategoryId(), item.getQuantity(), item.getUnitPrice()))
                        .toList()
        );
    }

    public record OrderItemView(Long productId, Long categoryId, int quantity, long unitPrice) {
    }
}

File: apps/kotlin/shopping-service/src/main/kotlin/com/booster/kotlin/shoppingservice/order/web/dto/response/OrderResponse.kt
package com.booster.kotlin.shoppingservice.order.web.dto.response

import com.booster.kotlin.shoppingservice.order.domain.Order
import com.booster.kotlin.shoppingservice.order.domain.OrderStatus

data class OrderResponse(
    val orderId: Long,
    val status: OrderStatus,
    val totalPrice: Long,
    val recipientName: String,
    val address1: String,
    val address2: String?,
    val items: List<OrderItemResponse>,
) {
    data class OrderItemResponse(
        val orderItemId: Long,
        val productName: String,
        val variantSku: String,
        val unitPrice: Long,
        val quantity: Int,
        val totalPrice: Long,
    )

    companion object {
        fun from(order: Order) = OrderResponse(
            orderId = order.id,
            status = order.status,
            totalPrice = order.totalPrice,
            recipientName = order.recipientName,
            address1 = order.address1,
            address2 = order.address2,
            items = order.items.map {
                OrderItemResponse(
                    orderItemId = it.id,
                    productName = it.productName,
                    variantSku = it.variantSku,
                    unitPrice = it.unitPrice,
                    quantity = it.quantity,
                    totalPrice = it.totalPrice,
                )
            },
        )
    }
}
